lib/kafka/fetcher.rb

```diff
@@ -173,6 +192,7 @@ def fetch_batches
         max_bytes = @max_bytes_per_partition[topic]
 
         partitions.each do |partition, offset|
+          next if @pause_manager.paused?(topic, partition)
```

Comment:
PauseManager isn't thread safe, so this could result in very hard-to-trace bugs.